### PR TITLE
feat: update sfdx lwc jest

### PR DESCRIPTION
**Pull request by msrivastav13**
### What does this PR do?

Updates sfdx lwc jest to latest and migrates few jest tests using wire adaptor as per migration [guide](https://github.com/salesforce/wire-service-jest-util/blob/master/docs/migrating-from-version-2.x-to-3.x.md)

### What issues does this PR fix or reference?

Issue [256](https://github.com/trailheadapps/easy-spaces-lwc/issues/256)

#256 

## The PR fulfills these requirements:

[ ] Tests for the proposed changes have been added/updated.
[ ] Code linting and formatting was performed.

### Functionality Before

<insert gif and/or summary>

### Functionality After

<insert gif and/or summary>

### CODE REVIEW

File: es-space-mgmt/main/default/lwc/reservationHelperForm/__tests__/reservationHelperForm.test.js
Comment (msrivastav13) [inline es-space-mgmt/main/default/lwc/reservationHelperForm/__tests__/reservationHelperForm.test.js]: @pozil I had to add this in since after I started using the `0.13` version of `sfdx-lwc-jest`, for the component that has a child component involving imperative apex, I  received the undefined error in jest tests whenever imperative apex call was made. Not sure if this is a feature or a bug.

I had to manually add the mocks for it to pass.

Comment (pozil) [reply es-space-mgmt/main/default/lwc/reservationHelperForm/__tests__/reservationHelperForm.test.js]: Thanks for taking care of this @msrivastav13.
@jodarove what's your take on this, is that expected behavior?

Comment (jodarove) [reply es-space-mgmt/main/default/lwc/reservationHelperForm/__tests__/reservationHelperForm.test.js]: @pozil, i don't think this is the expected behavior; @msrivastav13, (if possible) let me take a look before you merge so I can determine why this apex method is not being auto-mocked.

Comment (msrivastav13) [reply es-space-mgmt/main/default/lwc/reservationHelperForm/__tests__/reservationHelperForm.test.js]: Thanks, @jodarove! Looking forward to seeing what you find!

Comment (jodarove) [reply es-space-mgmt/main/default/lwc/reservationHelperForm/__tests__/reservationHelperForm.test.js]: created https://github.com/salesforce/sfdx-lwc-jest/issues/230

Comment (msrivastav13) [reply es-space-mgmt/main/default/lwc/reservationHelperForm/__tests__/reservationHelperForm.test.js]: @pozil Just FYI, there will be a new version once this bug is patched, and will have to upgrade all the sample apps with the newer version.

Also looks like now the imperative apex calls from LWC will be auto-mocked by the lib, so might be some more work to incorporate them as well.

Comment (jodarove) [reply es-space-mgmt/main/default/lwc/reservationHelperForm/__tests__/reservationHelperForm.test.js]: @msrivastav13, @pozil the issue is fixed (v0.12.6 (summer21) v0.14.0 (next). however, we had to remove the auto-mocks for apex and apexContinuation modules (https://github.com/salesforce/sfdx-lwc-jest/pull/234) because of [this issue](https://github.com/salesforce/sfdx-lwc-jest/issues/232).

Comment (pozil) [reply es-space-mgmt/main/default/lwc/reservationHelperForm/__tests__/reservationHelperForm.test.js]: @jodarove too bad we couldn't keep it. I understand that we have to revert to the using `registerApexTestWireAdapter` however, I'm seeing a deprecation warning if I do that in v0.12.6 or v0.14.0:
```
console.warn
    registerApexTestWireAdapter is deprecated. More details: https://github.com/salesforce/wire-service-jest-util/blob/master/docs/migrating-from-version-2.x-to-3.x.md
```

Did I miss something?

Comment (msrivastav13) [reply es-space-mgmt/main/default/lwc/reservationHelperForm/__tests__/reservationHelperForm.test.js]: @jodarove Please note that `0.14.0` breaks the tests for the wire adapter change.  I was hoping I have to only auto mock for the imperative apex and not for the wire adapters. Is it a bug or I am missing something?

File: es-space-mgmt/main/default/lwc/customerList/__tests__/customerList.test.js
Comment (jodarove) [inline es-space-mgmt/main/default/lwc/customerList/__tests__/customerList.test.js:20]: @msrivastav13 the platform does not auto-mock apex methods (see my [previous comment](https://github.com/trailheadapps/easy-spaces-lwc/pull/257#discussion_r661826190)), this module will need to be mocked by test authors. One way of doing this is:

```js

jest.mock(
    '@salesforce/apex/reservationManagerController.getCustomerList',
    () => {
        const { createApexTestWireAdapter } = require('@salesforce/sfdx-lwc-jest');
        return {
            default: createApexTestWireAdapter(jest.fn().mockImplementation(() => Promise.resolve())),
        };
    },
    { virtual: true }
);
```

File: es-space-mgmt/main/default/lwc/reservationHelperForm/__tests__/reservationHelperForm.test.js
Comment (jodarove) [reply es-space-mgmt/main/default/lwc/reservationHelperForm/__tests__/reservationHelperForm.test.js]: @pozil, I added a note on the [migration guide](https://github.com/salesforce/wire-service-jest-util/blob/master/docs/migrating-from-version-2.x-to-3.x.md#faq). Is not that we reverted to `registerApexTestWireAdapter` (which is still deprecated), is that the platform no longer auto-mocks apex methods (to a valid ApexTestWireAdapter). In order to remove the `registerApexTestWireAdapter` calls, test authors will need to mock their apex methods. cc/ @pmdartus

Comment (jodarove) [reply es-space-mgmt/main/default/lwc/reservationHelperForm/__tests__/reservationHelperForm.test.js]: @msrivastav13 can you elaborate more on breaks "wire adapter change" (check [this comment](https://github.com/trailheadapps/easy-spaces-lwc/pull/257#discussion_r662439382)).

> I was hoping I have to only auto mock for the imperative apex and not for the wire adapters

When you mock "imperative apex" you also mock the "wire adapters"; they are both the same module.

Comment (msrivastav13) [reply es-space-mgmt/main/default/lwc/reservationHelperForm/__tests__/reservationHelperForm.test.js]: ah! ok, I was assuming that for `@wire` decorated apex I do not need to do anything and it is auto-mocked which it seems is not possible without actually mocking it.

File: es-space-mgmt/main/default/lwc/customerList/__tests__/customerList.test.js
Comment (msrivastav13) [reply es-space-mgmt/main/default/lwc/customerList/__tests__/customerList.test.js:20]: @jodarove I simplified it to be like below
```js

jest.mock(
    '@salesforce/apex/reservationManagerController.getCustomerList',
    () => {
        const { createApexTestWireAdapter } = require('@salesforce/sfdx-lwc-jest');
        return {
            default: createApexTestWireAdapter(jest.fn()),
        };
    },
    { virtual: true }
);

```

It worked fine. 

If you do not mind could you review the commits [here](https://github.com/trailheadapps/easy-spaces-lwc/pull/257/commits/e2e8f9397285a2015089f1880692d2e0af344be5)

**Review by jodarove — APPROVED** [review]
lgtm

File: es-space-mgmt/main/default/lwc/customerList/__tests__/customerList.test.js
Comment (msrivastav13) [reply es-space-mgmt/main/default/lwc/customerList/__tests__/customerList.test.js:20]: @pozil Will merge once I get approval from you. 

We might want to broadcast this to a larger audience (maybe Twitter or stackexchange or Salesforce Developers Trailblazer Community)as they will be in the same boat if they are upgrading to the newer version.

Also whoever wrote this [module](https://trailhead.salesforce.com/content/learn/modules/test-lightning-web-components/write-a-jest-test-for-wire-service) on the trailhead, needs to be notified I guess.

**Review by pozil — APPROVED** [review]
Thanks @jodarove and @msrivastav13 for looking into this. Good to go!

**Pull request merged**
